Automattic/newspack-theme
# Single Post: First pass at adding featured image position option, to display behind title

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds an option to the featured image area of posts to display the featured image behind the post title.

When select, the post title, post meta (author name and published date) and share buttons should display over the image. To reduce visual noise, I've also hidden the author avatar.

The styles managing this still need a bit of fine tuning -- right now the image is set to be 80% of the screen size on desktop and shorter on smaller screens, but I'd like to explore this a bit more in future commits to optimize behaviour for different image sizes (I need to do further testing with a full-sized monitor). 

In the meantime, I'd love some feedback on how it works/looks, and anything that seems weird/unexpected. 

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Create a new post; confirm you have an option for featured image position, set to 'Default':

<img width="266" alt="image" src="https://user-images.githubusercontent.com/177561/64878956-c0991200-d622-11e9-845f-c4d559432312.png">

3. Add a featured image; right now the original 'too-small-image' fallback behaviour still applies, so images smaller than 1200px will be placed in the content area. Make sure to pick a larger image, and if this seems 'wrong', let me know!
4. Select 'Behind Article Title' for the image positioning.
5. Add a post title, and any other testing content you'd like.
6. Publish the post.
7. View on the front-end; make sure your featured image is now displaying behind the content, and looks roughly like this:

<img width="1427" alt="image" src="https://user-images.githubusercontent.com/177561/64879057-0950cb00-d623-11e9-944d-e7b7ca046e24.png">

8. Try scaling down the screen and check for any 'gotchas' or weirdness. 

### Other information:
